### PR TITLE
Semgrep OSS vs Semgrep AppSec Platform refresh

### PR DESCRIPTION
**Pull request by s-santillan**
# Thanks for improving Semgrep Docs 😀

## It's that time of the year for our semi-annual refresh of the OSS vs AppSec Platform (Semgrep) docs

This approach divides our features into **core workflows** since we want users to know that Semgrep isn't just a "code scanning tool" but also does remediation, reporting, SBOM export, license blocking, etc.

The core workflows are taken from the [design team's research work](https://docs.google.com/presentation/d/1wtn5BsVZfYT-WsQuBPlC4TO3UVYAyArlk6WkgQhBD5U/edit#slide=id.p) which I feel is instrumental in how we communicate information to the AppSec practitioner.

The first diagram you'll see displays the boundaries of Semgrep OSS as opposed to AppSec Platform.

Here's the [preview page](https://deploy-preview-1772--semgrep-docs-prod.netlify.app/docs/semgrep-pro-vs-oss) but you can just skip to the [Comparison of core workflows](https://deploy-preview-1772--semgrep-docs-prod.netlify.app/docs/semgrep-pro-vs-oss#comparison-by-core-workflows) or the [Appendix](https://deploy-preview-1772--semgrep-docs-prod.netlify.app/docs/semgrep-pro-vs-oss#appendix) if you just wanted a feature breakdown.

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] Sara to verify that we now have reachability rules for all CVEs starting 2017.
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

**Comment by s-santillan** [comment]
3-col experiment (see following screenshot) doesn't work so well, so I'll be changing the layout.

3-col-layout-experiment

**Comment by s-santillan** [comment]
Update: I think the 2-col looks a lot better:

2-col-layout

File: docs/semgrep-pro-vs-oss.md [hunk 1]
```diff
@@ -1,207 +1,411 @@
...
+:::
+
+## Comparison by core workflows
+
+![Scope of each offering by core workflows](/img/security-program-workflows.svg) <br />
+_**Figure**. A typical AppSec security program's core workflows and the scope of out-of-the-box Semgrep OSS and Semgrep AppSec Platform features._
+
+### Deployment
+
+The process of integrating Semgrep into your developer and infrastructure workflows._
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:53]:
```suggestion
The process of integrating Semgrep into your developer and infrastructure workflows.
```

File: docs/semgrep-pro-vs-oss.md [hunk 2]
```diff
@@ -1,207 +1,411 @@
...
+
+<div class="col-2-grid" >
+<div> 
+
+##### Semgrep OSS
+
+Semgrep OSS runs in your local machine's CLI through the `semgrep scan` command.
+
+Deploying in bulk or at scale is manual: it can scan a remote repository by running a CI job but you must write and configure the CI job for each repository.
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:62]:
```suggestion
Deploying in bulk or at scale is a manual task. Semgrep OSS can scan a remote repository by running as part of a CI job but you must write and configure the CI job for each repository. 
```

File: docs/semgrep-pro-vs-oss.md [hunk 3]
```diff
@@ -1,207 +1,411 @@
...
+
+<!-- Environments are omitted due to differences in licensing -
+`semgrep scan` can be run in IDE atm but the extension is proprietary -->
+
+</div>
+<div>
+
+##### Semgrep AppSec Platform 
+
+Semgrep AppSec Platform can perform scans in the following environments:
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:72]:
```suggestion
Semgrep AppSec Platform can scan in the following environments:
```

File: docs/semgrep-pro-vs-oss.md [hunk 4]
```diff
@@ -1,207 +1,411 @@
...
+
+- CI
+- Web app (for Managed Scans)
+- CLI
+- IDE
+- `pre-commit`
+
+Your scan configuration, such as rules and policies, and scan analysis (SAST, SCA, or secrets) are preserved across all environments.
+
+Users comfortable with granting Semgrep code access, can quickly deploy Semgrep to thousands of repositories through [Managed Scans](/deployment/managed-scanning).
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:82]:
```suggestion
Users comfortable with granting Semgrep code access can quickly deploy Semgrep to thousands of repositories through [Managed Scans](/deployment/managed-scanning).
```

File: docs/semgrep-pro-vs-oss.md [hunk 5]
```diff
@@ -1,207 +1,411 @@
...
+
+##### Semgrep OSS
+
+Semgrep OSS provides the following SAST analyses:
+
+- Single file, cross function constant propagation
+- Single function taint analysis 
+- Semantic analysis
+
+The scope makes it fast, at the cost of coverage and precision.
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:104]:
```suggestion
The limited scope makes it fast, at the cost of coverage and precision.
```

File: docs/semgrep-pro-vs-oss.md [hunk 6]
```diff
@@ -1,207 +1,411 @@
...
+</div>
+<div>
+
+##### Semgrep AppSec Platform 
+
+
+Semgrep AppSec Platform supports SAST, SCA, and secret scans as listed in [Product terms](#product-terms). You can run these **scan types** across all of your environments, preserving any configuration you have made.
+
+<details>
+<summary>Click to view Semgrep Code analyses (SAST)</summary>
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:117]:
```suggestion
<summary>Click to view Semgrep Code analyses and functions (SAST)</summary>
```

Not sure if this omission was deliberate or not, but if it is, feel free to ignore this comment

File: docs/semgrep-pro-vs-oss.md [hunk 7]
```diff
@@ -1,207 +1,411 @@
...
+
+
+<details>
+<summary>Click to view Semgrep Secrets analyses and functions</summary>
+- Validation of active, leaked secrets
+- Entropy
+- Historical scanning
+</details>
+
+Additionally, the Semgrep team maintains and contributes to premium rules, known as Pro rules, specifically making use of the advanced analyses listed here.
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:139]:
```suggestion
Additionally, the Semgrep team maintains and contributes to premium rules, known as Pro rules, that specifically making use of the advanced analyses listed here.
```

File: docs/semgrep-pro-vs-oss.md [hunk 8]
```diff
@@ -1,207 +1,411 @@
...
+
+</div>
+</div>
+
+<br />
+
 :::tip
-All Semgrep Pro products are free for up to 10 contributors.
+Certain languages, such as Apex, are available only on Semgrep AppSec Platform.
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:147]: Semgrep AppSec Platform or Semgrep Pro?

File: docs/semgrep-pro-vs-oss.md [hunk 9]
```diff
@@ -1,207 +1,411 @@
...
+
+Developers and AppSec engineers are able to provide reasons for a finding's status, such as **Acceptable risk** or **False positive** for **Ignored** findings.
+
+Semgrep AppSec Platform provides AI-assisted triage through Semgrep Assistant, which can analyze all your findings to suggest which findings it thinks are false positives. [<i class="fas fa-external-link fa-xs"></i> Learn more about Semgrep Assistant's accuracy.](https://semgrep.dev/blog/2023/assistant-public-beta)
+Assistant also provides AI-assisted remediation with the following features:
+
+- Step-by-step remediation.
+- Can be viewed by developers and AppSec engineers in their preferred environment.
+- Ability to learn your preferred libraries and functions through Assistant Memories.
+-
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:194]:
```suggestion

```

File: docs/semgrep-pro-vs-oss.md [hunk 10]
```diff
@@ -1,207 +1,411 @@
...
+_Tuning refers to the improvement of Semgrep's engine, rules, and policies to improve such metrics as the true positive rate, net new findings, and findings fixed before they enter production._
+
+_Tuning assists in the prevention of vulnerabilities from entering production._
+
+<div class="col-2-grid"> 
+<div>
+
+##### Semgrep OSS
+
+Tuning is not supported in Semgrep OSS but you can customize the rules you run on your scans.
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:211]:
```suggestion
Tuning is not supported in Semgrep OSS, but you can customize the rules you run on your scans.
```

File: docs/semgrep-pro-vs-oss.md [hunk 11]
```diff
@@ -1,207 +1,411 @@
...
+Semgrep OSS does not provide any metrics that may inform you of potential performance improvements you can make.
+
+</div>
+<div>
+
+##### Semgrep AppSec Platform
+
+The [Policies](/semgrep-code/policies) feature manages rules, helps block PRs or MRs from entering production, and configures which findings are presented to developers. This feature is available for both Semgrep Code and Secrets.
+
+You are able to test a rule's performance by first **monitoring** its performance (and showing it only in AppSec environments), then changing its mode to leave comments or help block a PR or MR from merging.
```
Comment (khorne3) [inline docs/semgrep-pro-vs-oss.md:222]:
```suggestion
You can test a rule's performance by first **monitoring** its performance (and showing it only in AppSec environments), then changing its mode to leave comments or help block a PR or MR from merging.
```

**Review by khorne3 — APPROVED** [review]
Just some minor nits

File: docs/semgrep-pro-vs-oss.md (hunk 8 above)
Comment (s-santillan) [reply docs/semgrep-pro-vs-oss.md:146]: Semgrep Pro is usually used mostly to refer to the rules (these days). Apex is part of the proprietary engine itself, so "Semgrep AppSec Platform" it is.

File: docs/semgrep-pro-vs-oss.md (hunk 7 above)
Comment (s-santillan) [reply docs/semgrep-pro-vs-oss.md:138]: "that specifically make" is the correction I went with.

File: docs/semgrep-pro-vs-oss.md (hunk 6 above)
Comment (s-santillan) [reply docs/semgrep-pro-vs-oss.md:117]: The Semgrep Code section was indeed the only product that was purely about SAST analyses - Dependency search in SCA isn't really an analysis, more a search function and Historical scans in Secrets is also a function, not an analysis.

**Comment by s-santillan** [comment]
This PR is considered done. 
Part of the request is to broaden our definition from GPT4 to AI.
However, I'll address the GPT grep-and-remove in a separate PR because it includes legal.

**Pull request merged**